NextcloudTalk/FileMessageTableViewCell.m

```diff
@@ -206,24 +218,43 @@ - (void)setupForMessage:(NCChatMessage *)message withLastCommonReadMessage:(NSIn
-                                               width = maxPreviewImageWidth;
+                                           if (width > previewMaxWidth) {
+                                               CGFloat ratio = previewMaxWidth / width;
+                                               width = previewMaxWidth;
                                                height = height * ratio;
                                            }
                                            weakSelf.vPreviewSize[3].constant = height;
                                            weakSelf.hPreviewSize[3].constant = width;
                                            weakSelf.vGroupedPreviewSize[1].constant = height;
                                            weakSelf.hGroupedPreviewSize[1].constant = width;
+                                           // if the image is very narrow, use a very small corner radius
+                                           if (height < 2 * kFileMessageCellMediaFilePreviewCornerRadius || width < 2 * kFileMessageCellMediaFilePreviewCornerRadius) {
+                                               weakSelf.previewImageView.layer.cornerRadius = MIN(height, width) / 2.0;
+                                           }
+                                           // use a bigger corner radius for media file previews since they're shown bigger
+                                           else {
+                                               weakSelf.previewImageView.layer.cornerRadius = isMediaFile ?  kFileMessageCellMediaFilePreviewCornerRadius : kFileMessageCellFilePreviewCornerRadius;
+                                           }
+                                           // only show the play icon if there is an image preview (not on top of the default video placeholder)
+                                           weakSelf.playIconImageView.hidden = !isVideoFile;
+                                           if (isVideoFile) {
+                                               // if the video preview is very narrow, make the play icon fit inside
+                                               weakSelf.playIconImageView.frame = CGRectMake(0, 0, MIN(MIN(height, width), kFileMessageCellVideoPlayIconSize), MIN(MIN(height, width), kFileMessageCellVideoPlayIconSize));
+                                               weakSelf.playIconImageView.center = CGPointMake(width / 2.0, height / 2.0);
+                                           }
                                            [weakSelf.previewImageView setImage:image];
                                            [weakSelf setNeedsLayout];
                                            [weakSelf layoutIfNeeded];
-                                            
+                                           
                                            if (weakSelf.delegate) {
                                                [weakSelf.delegate cellHasDownloadedImagePreviewWithHeight:height forMessage:message];
```

Comment:
```suggestion
                                               [weakSelf.delegate cellHasDownloadedImagePreviewWithHeight:ceil(height) forMessage:message];
```

Because the height can be a fraction, we should use `ceil` here (it helps later, see other changes)